trio/tests/test_exports.py

```diff
@@ -114,6 +143,169 @@ def no_underscores(symbols):
+                missing.remove("add_note")
+
+            # TODO: why is this? Is it a problem?
+            if class_ == trio.StapledStream:
+                extra.remove("receive_stream")
+                extra.remove("send_stream")
+
+            if missing | extra:
+                errors[f"{module_name}.{class_name}"] = {
+                    "missing": missing,
+                    "extra": extra,
+                }
+        elif tool == "mypy":
+            tmpfile = tmpdir / "check_mypy.py"
+            sorted_runtime_names = list(sorted(runtime_names))
+            content = f"from {module_name} import {class_name}\n" + "".join(
+                f"{class_name}.{name}\n" for name in sorted_runtime_names
+            )
+            tmpfile.write_text(content, encoding="utf8")
+            from mypy.api import run
+
+            res = run(
+                [
+                    "--config-file=",
+                    "--follow-imports=silent",
+                    "--disable-error-code=operator",
+                    str(tmpfile),
+                ]
+            )
+            if res[2] != 0:
+                print(res)
+                it = iter(res[0].split("\n")[:-2])
+                for output_line in it:
+                    # hack to get around windows path starting with <drive>:<path>
+                    output_line = output_line[2:]
+
+                    _, line, error_type, message = output_line.split(":")
+
+                    # -2 due to lines being 1-indexed and to skip the import line
+                    symbol = (
+                        f"{module_name}.{class_name}."
+                        + sorted_runtime_names[int(line) - 2]
+                    )
+
+                    # The POSIX-only attributes get listed in `dir(trio.Path)` since
+                    # they're in `dir(pathlib.Path)` on win32 cpython. This should *maybe*
+                    # be fixed in the future, but for now we ignore it.
+                    if (
+                        symbol
+                        in ("trio.Path.group", "trio.Path.owner", "trio.Path.is_mount")
+                        and sys.platform == "win32"
+                        and sys.implementation.name == "cpython"
+                    ):
+                        continue
+
+                    # a bunch of symbols have this error, e.g. trio.lowlevel.Task.context
+                    # but I don't think that's a problem - and if it is it would likely
+                    # be picked up by "proper" mypy tests elsewhere
+                    if "conflicts with class variable access" in message:
+                        continue
+
+                    errors[symbol] = error_type + ":" + message
+
+        else:  # pragma: no cover
+            assert False
+
+    if errors:  # pragma: no cover
+        from pprint import pprint
+
+        print(f"\n{tool} can't see the following symbols in {module_name}:")
```

Comment:
Do we want a bigger separator? Like, `\n\n--------\n{tool} can't see the following symbols in {module_name}:`? IDK

Reply:
On second thought, why do we need to pprint `errors`? Shouldn't `pytest` do that for us with `assert not errors`?